src/internal_rules.jl

```diff
@@ -761,47 +775,73 @@ function EnzymeRules.augmented_primal(
             end
         end
     end
-    cache = if isa(A, Const)
-        nothing
-    else
-        dfact
-    end
 
+    cache = isa(A, Const) ? nothing : (fact_returned, dfact)
```

Comment:
This one should just be fact (aka this is not the returned one, which is a better name for the one below).